src/jsonSchema.ts

```diff
@@ -0,0 +1,98 @@
+import { appendErrors, Resolver, transformToNestObject } from 'react-hook-form';
+//import * as Ajv from 'ajv';
+import Ajv, { ErrorObject, Options } from 'ajv';
+import type { JSONSchema4, JSONSchema6, JSONSchema7 } from 'json-schema';
+import {
+  FieldError,
+  FieldErrors,
+  ResolverError,
+  ResolverSuccess,
+} from 'react-hook-form/dist/types/form';
+
+export type JSONSchema = JSONSchema4 | JSONSchema6 | JSONSchema7;
+
+const parseErrorSchema = <TFieldValues extends Record<string, any>>(
+  validationError: Array<ErrorObject> | null | undefined,
+  validateAllFieldCriteria: boolean,
+): FieldErrors<TFieldValues> =>
+  Array.isArray(validationError)
+    ? validationError.reduce(
+        (
+          previous: FieldErrors<TFieldValues>,
+          { dataPath, keyword, message },
+        ) => {
+          const path = dataPath.replace(/\//g, '.').replace(/^\./, '');
```

Comment:
I probably need to run it to understand better, so their path shape is different?

if so perhaps it's better to move this function outside the scope of the `reduce`.

Reply:
Removed the first `replace` call. Change already committed.